apis/externalsecrets/v1alpha1/pushsecret_types.go

```diff
@@ -231,12 +236,116 @@ type PushSecret struct {
 	Status PushSecretStatus `json:"status,omitempty"`
 }
 
+// PushSecretList contains a list of PushSecret resources.
 // +kubebuilder:object:root=true
 // +kubebuilder:printcolumn:name="AGE",type="date",JSONPath=".metadata.creationTimestamp"
 // +kubebuilder:printcolumn:name="Status",type=string,JSONPath=`.status.conditions[?(@.type=="Ready")].reason`
-// PushSecretList contains a list of PushSecret resources.
 type PushSecretList struct {
 	metav1.TypeMeta `json:",inline"`
 	metav1.ListMeta `json:"metadata,omitempty"`
 	Items           []PushSecret `json:"items"`
 }
+
+// ClusterPushSecretCondition used to refine PushSecrets to specific namespaces and names.
+type ClusterPushSecretCondition struct {
+	// Choose namespace using a labelSelector
+	// +optional
+	NamespaceSelector *metav1.LabelSelector `json:"namespaceSelector,omitempty"`
+
+	// Choose namespaces by name
+	// +optional
+	// +kubebuilder:validation:items:MinLength:=1
+	// +kubebuilder:validation:items:MaxLength:=63
+	// +kubebuilder:validation:items:Pattern:=^[a-z0-9]([-a-z0-9]*[a-z0-9])?$
+	Namespaces []string `json:"namespaces,omitempty"`
+}
+
+// PushSecretMetadata defines metadata fields for the PushSecret generated by the ClusterPushSecret.
+type PushSecretMetadata struct {
+	// +optional
+	Annotations map[string]string `json:"annotations,omitempty"`
+
+	// +optional
+	Labels map[string]string `json:"labels,omitempty"`
+}
+
+type ClusterPushSecretSpec struct {
+	// PushSecretSpec defines what to do with the secrets.
+	PushSecretSpec PushSecretSpec `json:"pushSecretSpec"`
+	// The time in which the controller should reconcile its objects and recheck namespaces for labels.
+	RefreshInterval *metav1.Duration `json:"refreshTime,omitempty"`
+	// The name of the push secrets to be created.
+	// Defaults to the name of the ClusterPushSecret
+	// +optional
+	// +kubebuilder:validation:MinLength:=1
+	// +kubebuilder:validation:MaxLength:=253
+	// +kubebuilder:validation:Pattern:=^[a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*$
+	PushSecretName string `json:"pushSecretName,omitempty"`
+
+	// The metadata of the external secrets to be created
+	// +optional
+	PushSecretMetadata PushSecretMetadata `json:"pushSecretMetadata,omitempty"`
+
+	// A list of labels to select by to find the Namespaces to create the ExternalSecrets in. The selectors are ORed.
+	// +optional
+	NamespaceSelectors []*metav1.LabelSelector `json:"namespaceSelectors,omitempty"`
+
+	// Choose namespaces by name
+	// +optional
+	// +kubebuilder:validation:items:MinLength:=1
+	// +kubebuilder:validation:items:MaxLength:=63
+	// +kubebuilder:validation:items:Pattern:=^[a-z0-9]([-a-z0-9]*[a-z0-9])?$
+	Namespaces []string `json:"namespaces,omitempty"`
```

Comment:
While we do have this field for ClusterExternalSecret, I think it makes no sense to add it for ClusterPushSecret as this can be achieved with a namespace selector matching a label `app.kubernetes.io/namespace-name: my-namespace`

Reply:
👍